source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -115,13 +139,56 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   return Http::FilterHeadersStatus::StopIteration;
 }
 
+Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers, bool) {
+  const auto* config = getConfig();
+
+  if (config->enabled() && config->enableXRateLimitHeaders()) {
+    ASSERT(stored_descriptors_.has_value());
+    auto limit = maxTokens(stored_descriptors_.value());
+    auto remaining = remainingTokens(stored_descriptors_.value());
+    auto reset = remainingFillInterval(stored_descriptors_.value());
+    stored_descriptors_.reset();
+
+    headers.addCopy(HttpFilters::Common::RateLimit::XRateLimitHeaders::get().XRateLimitLimit,
+                    limit);
+    headers.addCopy(HttpFilters::Common::RateLimit::XRateLimitHeaders::get().XRateLimitRemaining,
+                    remaining);
+    headers.addCopy(HttpFilters::Common::RateLimit::XRateLimitHeaders::get().XRateLimitReset,
+                    reset);
```

Comment:
we want `addReferenceKey()` for all of these calls, no need to copy the keys.

Reply:
Thanks, I will change them.